apps/condo/domains/miniapp/schema/SendB2BAppPushMessageService.js

```diff
@@ -0,0 +1,218 @@
+                    `sendB2BAppPushMessage:${type}:app:${b2bAppFilter.id}:org:${organizationFilter.id}:user:${userFilter.id}`,
+                    get(appSettings, 'notificationWindowSize') || DEFAULT_NOTIFICATION_WINDOW_DURATION_IN_SECONDS,
+                    get(appSettings, 'numberOfNotificationInWindow') || DEFAULT_NOTIFICATION_WINDOW_MAX_COUNT,
+                    context,
+                )
+
+                const [b2bAppContext] = await itemsQuery('B2BAppContext', {
+                    where: {
+                        organization: { ...organizationFilter, deletedAt: null },
+                        app: { ...b2bAppFilter, deletedAt: null },
+                        status: CONTEXT_FINISHED_STATUS,
+                        deletedAt: null,
+                    },
+                    first: 1,
+                })
+                if (!b2bAppContext) {
+                    throw new GQLError(ERRORS.NO_B2B_CONTEXT, context)
+                }
+
+                const [b2bAppAccessRight] = await itemsQuery('B2BAppAccessRight', {
+                    where: {
+                        accessRightSet: {
+                            app: { ...b2bAppFilter, deletedAt: null },
+                            canExecuteSendB2BAppPushMessage: true,
+                            deletedAt: null,
+                        },
+                        user: { id: authedItemId, deletedAt: null },
+                        deletedAt: null,
+                    },
+                    first: 1,
+                })
+                if (!b2bAppAccessRight) {
+                    throw new GQLError(ERRORS.NO_B2B_APP_ACCESS_RIGHT, context)
+                }
+
+                const [organization] = await itemsQuery('Organization', {
+                    where: {
+                        ...organizationFilter,
+                        deletedAt: null,
+                    },
+                    first: 1,
+                })
+
+                const [employee] = await itemsQuery('OrganizationEmployee', {
+                    where: {
+                        organization: { ...organizationFilter, deletedAt: null },
+                        user: { ...userFilter, deletedAt: null },
+                        isAccepted: true,
+                        isRejected: false,
+                        isBlocked: false,
+                        deletedAt: null,
+                    },
+                    first: 1,
+                })
+                if (!employee) {
+                    throw new GQLError(ERRORS.NO_EMPLOYEE_FOR_USER, context)
+                }
+
+                const roleId = get(employee, 'role', null)
+                const [b2bAppRole] = await itemsQuery('B2BAppRole', {
+                    where: {
+                        deletedAt: null,
+                        app: { ...b2bAppFilter, deletedAt: null },
+                        role: { id: roleId, deletedAt: null },
+                    },
+                    first: 1,
+                })
```

Comment:
maybe it’s worth adding data caching for a couple of minutes and adding RLock so that when requests are sent to all employees, only the first request goes to the database